GymApp/GymAppApi/Controllers/VisitorsController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using GYM.API.Data;
+using GYM.API.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace GYM.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class VisitorsController : ControllerBase
+    {
+        private readonly GymDbContext _context;
+
+        public VisitorsController(GymDbContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Visitors
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Visitor>>> GetVisitors()
+        {
+            return await _context.Visitors.ToListAsync();
+        }
+
+        // GET: api/Visitors/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Visitor>> GetVisitor(int id)
+        {
+            var visitor = await _context.Visitors.FindAsync(id);
+
+            if (visitor == null)
+            {
+                return NotFound();
+            }
+
+            return visitor;
+        }
+
+        // PUT: api/Visitors/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutVisitor(int id, Visitor visitor)
+        {
+            if (id != visitor.Id)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(visitor).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!VisitorExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
```

Comment:
Not sure that it needs for now, maybe better just to check if the entity exists.